src/protocols/arinc818.stanza

```diff
@@ -0,0 +1,160 @@
+connections between sources and receivers on a printed circuit board.
+@see https://docs.wixstatic.com/ugd/aefb08_37d298772ed9485f97af8f2954270e38.pdf?index=true
+<DOC>
+#use-added-syntax(jitx)
+defpackage jsl/protocols/arinc818:
+  import core
+  import jitx
+  import jitx/commands
+
+  import jsl/errors
+  import jsl/bundles
+  import jsl/pin-assignment
+  import jsl/si/helpers
+  import jsl/si/pairs
+  import jsl/si/constraints
+
+doc: \<DOC>
+@brief ARINC Type
+This enumerated type defines either a TX type or RX type for the ARINC818 bundle.
+<DOC>
+public pcb-enum jsl/protocols/arinc818/ARINC818Type :
+  TX
+  RX
+
+doc: \<DOC>
+@brief ARINC Generation enums
+This is a fixed list of possible generation definitions for ARINC818
+<DOC>
+
+public pcb-enum jsl/protocols/arinc818/ARINC818Version:
+  ARINC818-FC1x   ;   1.0625 Gbits/s
+  ARINC818-1p5    ;   1.5    Gbits/s
+  ARINC818-1p62   ;   1.62   Gbits/s
+  ARNIC818-FC2x   ;   2.125  Gbits/s
+  ARINC818-2p5    ;   2.5    Gbits/s
+  ARINC818-3p1875 ;   3.1875 Gbits/s
+  ARNIC818-FC4x   ;   4.5    Gbits/s
+  ARINC818-5p0    ;   5.0    Gbits/s
+  ARINC818-FC6x   ;   6.375  Gbits/s
+  ARINC818-FC8x   ;   8.5    Gbits/s
+  ARINC818-FC12x  ;  12.75   Gbits/s
+  ARINC818-FC16x  ;  14.025  Gbits/s
+  ARINC818-FC24x  ;  21.0375 Gbits/s
+  ARINC818-FC32x  ;  28.05   Gbits/s
+
+
+
+doc: \<DOC>
+@brief ARINC818 Bundle
+The Data Bundle consists of 1 differential pair (1 RX or TX)
+@member TX/RX for the ARINC818 bundle (consists of either a TX or RX diff pair)
+<DOC>
+
+public pcb-bundle ARINC818 (type:ARINC818Type) :
+  name = "ARINC818"
+  description = "ARINC 818 AVDB protocol"
+  switch(type) :
+    TX : make-port(`TX, diff-pair)
+    RX : make-port(`RX, diff-pair)
+
+doc: \<DOC>
+@brief Construct the ARINC818 Topology and Constraints
+
+This function constructs the ARINC818 topology and applies constraints
+to the channel for intra-pair skew and channel loss. Note that a connection
+needs to be between a TX on a transmitter and a RX on a receiver.
+
+@param sw The intra-pair skew value for the differential signals being connected.
+@param ml The maximum loss for all differential signals being connected.
+@param src This is expected to be a port of `Bundle` type `ARINC818`
+@param dst This is expected to be a port of `Bundle` type `ARINC818`
+
+<DOC>
+public defn connect-ARINC818 (cons:SI-Constraint, src:JITXObject, dst:JITXObject) :
+  inside pcb-module :
+    check-bundle(ARINC818(TX), src)
+    check-bundle(ARINC818(RX), dst)
+    check-correct-pins(src.TX.P, dst.RX.P)
+    within [s,d] = constrain-topology(src.TX => dst.RX, cons) :
+      topo-net(s d)
+
+doc: \<DOC>
+@brief Check the presence of a TX and RX version of the ARINC818 bundle
+
+This function checks the presence of the TX and RX pins on the ARINC818 bundle.
+
+@param p-tx The TX pin
+@param p-rx The RX pin
+<DOC>
+public defn check-correct-pins (p-tx:JITXObject p-rx:JITXObject):
```

Comment:
We have a `check-single-pin` already:

https://github.com/JITx-Inc/jsl/blob/6c9232ff401c2e14cc12158be732462d5221a37b/src/pin-assignment/checks.stanza#L9

Reply:
Yep, tried to use that first. But the error message would not make sense in the context that you did not connect the TX to RX but instead connected RX to TX. The reason to make this directional is that some implementations are optical and in that case, you need some more directionality?